### DISCUSSION TITLE
You are making Prodigy less secure.

### DISCUSSION THREAD

**Discussion by hostedposted** [ProdigyPNP/ProdigyMathGameHacking#87] [Developer Forum]
It is impossible to patch P-NP. It modifies the source code of Prodigy. You should probably change We aren't evil in the README.md. It should be something like We are evil

Firstly, you stole this code from PMGH.
Secondly, you're kind of making prodigy less secure.

**Comment by afkvido** [comment]
*Not **impossible** to patch P-NP, if they well, yknow, rewrote the inventory handler, and with that half the game.

*It doesn't modify Prodigy's actual source code- It modifies what your CLIENT sees as the source code

*We're not evil, I do this for **fun**

*I didn't **steal**. I'm reviving their project

*If Prodigy paid attention to the PROPER fix and not a cheap one, they'd be somewhat secure by now.


So yeah, Idk if ur sped or smth, but uhh yknow I don't think you know how ProdigyMathGameHacking works

**Reply by hostedposted** [reply]
1. P-NP auto patches the gamefile. Let's say prodigy makes it so items are received server side like the runes, that would still be exploitable. Just mass battle. Sure they have an endpoint for secure battles, but still look at my rune hacks.
2. It modifies the source code your client executes, that is implied.
3. Prodigy has been doing proper fixes. It's just that P-NP auto unpatches.

Well it makes sense that you think it is impossible to patch P-NP. You're not good enough at coding to be able to fix P-NP if prodigy patches it.

**Reply by afkvido** [reply]
P-NP unpactches little pieces of shit that Prodigy throws at us that they call "security".

You know what _would_ be great security? Them making items, most events, etc. all **s e r v e r s i d e**

**Reply by eris-webserv** [reply]
I'm going to point out that would be seriously annoying without making clientside communication hell

**Reply by afkvido** [reply]
xd. How about the client sends the event (ex. chest opened), the chest is verified with something like a UUID or SHA256 hash, and only then does the player recieve all rewards.

Include this for wild chests, monster battles, player hairstyle changes, etc.

**Reply by hostedposted** [reply]
So? We can find out that UUID and simply send the request ourselves.

**Reply by eris-webserv** [reply]
Change the UUID every 5 days, would also make our hack dev annoying as FUCK. Boom, no more hacks.

**Reply by afkvido** [reply]
> So? We can find out that UUID and simply send the request ourselves.

No. Generate a new chest UUID every time the player enters the zone. Unique for every player.

**Reply by eris-webserv** [reply]
That's a better idea.

**Reply by afkvido** [reply]
thanks :)

**Reply by hostedposted** [reply]
How would the client receive that UUID? Through a request. All requests can be remade. Also again, P-NP can intercept the request to get the uuid.

**Reply by afkvido** [reply]
Idea for prodigy: the client recieves the chest contents and location, the server has a UUID that represents that chest contents + location.

**Reply by afkvido** [reply]
> How would the client receive that UUID? Through a request. All requests can be remade. Also again, P-NP can intercept the request to get the uuid.

Even so, a hundred chests is nothing like 9 million gold
